File: frontend/src/services/apiDebug.ts
/**
 * AI 接口调试服务
 *
 * 后端路由前缀: /api/api-debug
 * 接口分组:
 *   1. 执行接口   - POST /execute
 *   2. AI 分析    - POST /analyze/{record_id}  POST /analyze/inline
 *   3. 健康检查   - GET  /health
 *   4. 记录查询   - GET  /records/list  GET /records/{id}  DELETE /records/{id}
 *   5. 错误模式   - GET  /patterns
 */
import request from './request';

// ============================================================
// 类型定义
// ============================================================

export type HttpMethod = 'GET' | 'POST' | 'PUT' | 'DELETE' | 'PATCH' | 'HEAD' | 'OPTIONS';

export interface ExecuteRequest {
  method: HttpMethod;
  url: string;
  headers?: Record<string, string>;
  params?: Record<string, any>;
  body?: any;
  auth?: {
    type: 'none' | 'bearer' | 'basic' | 'api_key';
    token?: string;
    username?: string;
    password?: string;
    key_name?: string;
    key_value?: string;
  };
  timeout?: number;
  api_id?: number;
  case_id?: number;
  env?: string;
  auto_analyze?: boolean;
}

export interface ExecuteResult {
  status: 'success' | 'failed' | 'error' | 'timeout' | 'pending';
  record_id: number;
  request: {
    method: string;
    url: string;
    headers: Record<string, string>;
    params: Record<string, any>;
    body: any;
    auth?: any;
  };
  response?: {
    status_code: number;
    headers: Record<string, string>;
    body: any;
    size: number;
  };
  status_code?: number;
  duration: number;
  error?: string;
  analysis?: AnalysisResult;
}

export interface AnalysisResult {
  status: 'success' | 'degraded' | 'error' | 'skipped';
  record_id?: number;
  problem_cause: string;
  solution: string;
  fix_suggestion: string;
  confidence: number;
  category?: string;
  source: 'llm' | 'rule_engine';
  elapsed_ms: number;
  message?: string;
}

export interface ExecutionRecord {
  id: number;
  api_id?: number;
  case_id?: number;
  execution_id?: number;
  method: string;
  url: string;
  headers?: any;
  params?: any;
  body?: any;
  auth?: any;
  status_code?: number;
  response_headers?: any;
  response_body?: any;
  response_size?: number;
  status: string;
  duration: number;
  error?: string;
  analysis_status: string;
  analysis_result?: any;
  analyzed_at?: string;
  env?: string;
  trigger_source?: string;
  client_ip?: string;
  created_at?: string;
}

export interface ExecutionRecordList {
  total: number;
  page: number;
  page_size: number;
  items: ExecutionRecord[];
}

export interface DebugHealth {
  status: string;
  agent: string;
  capabilities: string[];
  patterns_count: number;
  llm_available: boolean;
  supported_actions: string[];
}

export interface ErrorPattern {
  id: string;
  keywords: string[];
  problem_cause: string;
  confidence: number;
}

// ============================================================
// API 调用
// ============================================================

/** 执行 HTTP 请求 */
export async function executeRequest(data: ExecuteRequest): Promise<ExecuteResult> {
  const resp = await request.post('/api-debug/execute', data);
  return resp.data;
}

/** AI 分析执行记录 */
export async function analyzeRecord(recordId: number, force = false): Promise<AnalysisResult> {
  const resp = await request.post(`/api-debug/analyze/${recordId}`, null, {
    params: { force },
  });
  return resp.data;
}

/** 直接分析(不落库) */
export async function analyzeInline(payload: {
  request?: any;
  response?: any;
  error?: string;
  status?: string;
}): Promise<AnalysisResult> {
  const resp = await request.post('/api-debug/analyze/inline', payload);
  return resp.data;
}

/** Agent 健康检查 */
export async function debugHealthCheck(): Promise<DebugHealth> {
  const resp = await request.get('/api-debug/health');
  return resp.data;
}

/** 执行记录列表 */
export async function listExecutionRecords(params: {
  api_id?: number;
  case_id?: number;
  status?: string;
  status_code?: number;
  method?: string;
  keyword?: string;
  page?: number;
  page_size?: number;
}): Promise<ExecutionRecordList> {
  const resp = await request.get('/api-debug/records/list', { params });
  return resp.data;
}

/** 执行记录详情 */
export async function getExecutionRecord(recordId: number): Promise<ExecutionRecord> {
  const resp = await request.get(`/api-debug/records/${recordId}`);
  return resp.data;
}

/** 删除执行记录 */
export async function deleteExecutionRecord(recordId: number): Promise<void> {
  await request.delete(`/api-debug/records/${recordId}`);
}

/** 错误模式列表 */
export async function listErrorPatterns(): Promise<{ total: number; patterns: ErrorPattern[] }> {
  const resp = await request.get('/api-debug/patterns');
  return resp.data;
}

File: backend/app/schemas/api_debug.py
"""
API 执行记录 + AI 调试 Schema

定义:
  - ExecuteRequest:        执行接口请求
  - ExecutionRecordResponse: 执行记录响应
  - AnalyzeRequest:        AI 分析请求
  - AnalysisResult:        分析结果
"""
from typing import Any, Dict, List, Optional, Union
from datetime import datetime
from pydantic import BaseModel, Field


# ============================================================
# 执行接口
# ============================================================

class ExecuteRequest(BaseModel):
    """执行接口请求"""
    method: str = Field(..., description="HTTP 方法: GET/POST/PUT/DELETE/PATCH")
    url: str = Field(..., description="请求 URL")
    headers: Optional[Dict[str, str]] = Field(default=None, description="请求头")
    params: Optional[Dict[str, Any]] = Field(default=None, description="Query 参数")
    body: Optional[Any] = Field(default=None, description="请求体")
    auth: Optional[Dict[str, Any]] = Field(default=None, description="认证信息")
    timeout: int = Field(default=30000, ge=1000, le=120000, description="超时(ms)")
    api_id: Optional[int] = Field(default=None, description="关联接口 ID")
    case_id: Optional[int] = Field(default=None, description="关联用例 ID")
    env: str = Field(default="test", description="执行环境")
    auto_analyze: bool = Field(default=False, description="执行失败后自动调用 AI 分析")


class ExecuteResponse(BaseModel):
    """执行接口响应"""
    status: str = Field(..., description="success/failed/error/timeout")
    record_id: Optional[int] = Field(None, description="执行记录 ID")
    request: Dict[str, Any]
    response: Optional[Dict[str, Any]] = None
    status_code: Optional[int] = None
    duration: float = Field(0, description="耗时(ms)")
    error: Optional[str] = None
    analysis: Optional[Dict[str, Any]] = Field(None, description="AI 分析结果(仅 auto_analyze=True 且失败时)")


# ============================================================
# 执行记录
# ============================================================

class ExecutionRecordResponse(BaseModel):
    """执行记录响应"""
    id: int
    api_id: Optional[int] = None
    case_id: Optional[int] = None
    execution_id: Optional[int] = None
    method: str
    url: str
    headers: Optional[Any] = None
    params: Optional[Any] = None
    body: Optional[Any] = None
    auth: Optional[Any] = None
    status_code: Optional[int] = None
    response_headers: Optional[Any] = None
    response_body: Optional[Any] = None
    response_size: Optional[int] = None
    status: str
    duration: float
    error: Optional[str] = None
    analysis_status: str = "none"
    analysis_result: Optional[Any] = None
    analyzed_at: Optional[str] = None
    env: Optional[str] = None
    trigger_source: Optional[str] = None
    client_ip: Optional[str] = None
    created_at: Optional[str] = None


class ExecutionRecordListResponse(BaseModel):
    """执行记录列表"""
    total: int
    page: int
    page_size: int
    items: List[ExecutionRecordResponse]


# ============================================================
# AI 分析
# ============================================================

class AnalyzeRequest(BaseModel):
    """AI 分析请求"""
    record_id: Optional[int] = Field(None, description="执行记录 ID(二选一)")
    force: bool = Field(default=False, description="强制分析(即使请求成功)")
    request: Optional[Dict[str, Any]] = Field(None, description="直接传入请求(二选一)")
    response: Optional[Dict[str, Any]] = Field(None, description="直接传入响应")
    error: Optional[str] = Field(None, description="直接传入错误信息")
    status: Optional[str] = Field(None, description="执行状态")


class AnalysisResult(BaseModel):
    """AI 分析结果"""
    status: str
    record_id: Optional[int] = None
    problem_cause: str
    solution: str
    fix_suggestion: str
    confidence: float
    category: Optional[str] = None
    source: str = Field(..., description="llm / rule_engine")
    elapsed_ms: int = 0


class HealthResponse(BaseModel):
    """健康检查"""
    status: str
    agent: str
    capabilities: List[str]
    patterns_count: int
    llm_available: bool
    supported_actions: List[str]
